src/manifest.c

```diff
@@ -538,12 +559,30 @@ static int retrieve_manifests(int current, int version, char *component, struct
 	free(dir);
 
 	/* FILE is not set for a MoM, only for bundle manifests */
-	if (file && current < version) {
+	if (file && current < version && strcmp(component, "full") != 0) {
 		if (try_delta_manifest_download(current, version, component, file) == 0) {
 			return 0;
 		}
 	}
 
+	/* If it's mix content just hardlink instead of curl download */
+	if (is_mix) {
+		string_or_die(&filename, "%s/%i/Manifest.%s.tar", state_dir, version, component);
+		string_or_die(&url, "%s/%i/Manifest.%s.tar", basedir, version, component);
```

Comment:
This variable does not store a URL. Can you rename it to something better?

Reply:
That's arguable, it's technically storing a local file url, vs a regular http URL a few lines down if it fails. We have stack to spare, but I don't think it's too confusing to just use the same var. @matthewrsj thoughts?

Reply:
Honestly if it is just a file url I don't really care about the naming scheme. `url = "file://"` seems reasonable to me

Reply:
Maybe I'm misunderstanding the code, but `url` does not appear to have a `file://` prefix in this case. Are you suggesting that it should have a `file://` prefix?

Reply:
I will ignore my concern here for now. To me, storing a string representing a file path in a variable that implies it is a URL is confusing, but it is cosmetic when thinking about this PR as a whole.